src/components/ItemsBox.vue

```diff
@@ -15,14 +17,74 @@
 </template>
 
 <script>
+    import DisplayItemBoxInformation from './DisplayItemBoxInformation';
+
     export default {
         name: 'items-box',
         props: ['items'],
+        components: {
+            DisplayItemBoxInformation,
+        },
+        data: () => ({
+            dialogProp: false,
+        }),
+        methods: {
+            displayFields(item) {
+                if (item.__typename == 'Expense')
+                    return this.displayFieldsExpanse(item);
+                else if (item.__typename === 'Transaction')
+                    return this.displayFieldsTransaction(item);
+                else if (item.__typename === 'User')
+                    return this.displayFieldsUser(item);
+
+            },
+            displayFieldsExpanse(item) {
```

Comment:
prefer creating objects in-place. this will help you with a more functional thinking methodology. in here this would mean:
```
return [
  this.displayField('Name', item.name, 'string'),
  this.displayField('Date', item.date, 'date'),
  this.displayField('Price', item.price, 'string'),
  this.displayField('Payer', item.payer, 'avatar'),
  {
    fieldName: 'People',
    displayText: [...item.people],
    type: 'avatarList',
  }
];
```
// this does not include the other comment

Reply:
also, the code is shorter and please do this for all methods that create the display fields